stackit/internal/services/logme/instance/resource.go

```diff
@@ -299,6 +301,11 @@ func (r *instanceResource) Read(ctx context.Context, req resource.ReadRequest, r
 
 	instanceResp, err := r.client.GetInstance(ctx, projectId, instanceId).Execute()
 	if err != nil {
+		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+		if ok && (oapiErr.StatusCode == http.StatusNotFound) || (oapiErr.StatusCode == http.StatusGone) {
```

Comment:
`if ok && ((oapiErr.StatusCode == http.StatusNotFound) || (oapiErr.StatusCode == http.StatusGone))`

This is the same right? Because Go is lazy so I was wondering if it would check the first AND and ignore the last part

Reply:
Actually I think yes, but I meant to have the parenthesis wrapping the two last statements (for readability and to not rely on that prioritization of operations), good catch